Applications/Viqui/vqCore.cxx

```diff
@@ -3387,6 +3393,37 @@ void vqCore::generateReport(QString path, bool generateVideo)
   this->exportKml(path);
 }
 
+//-----------------------------------------------------------------------------
+void vqCore::exportSvm(QString path)
+{
+  QFile file(path);
+  if (!file.open(QIODevice::WriteOnly |
+                 QIODevice::Truncate))
+    {
+    qDebug() << file.errorString();
+    QMessageBox::warning(0, QString(), "Unable to write SVM file.");
+    return;
+    }
+
+  if (this->SavedQueryPlan.isSimilarityQuery())
```

Comment:
I'd prefer to disable the UI action entirely if the current query is not a similarity query. (It looks like that would be best accomplished by adding a `queryIqrModelAvailabilityChanged` signal?) Failing that, we should immediately (i.e. before asking for a file name) show a warning. Moving the code to get the file name into here (and having the UI action call this slot directly) will help.

Saving an empty file, especially with no explanation why, is not very useful.